lib/components/viewers/pattern-row.tsx

```diff
@@ -68,113 +72,63 @@ class PatternRow extends Component<Props, State> {
     }
 
     const routeName = route.shortName ? route.shortName : route.longName
+    const routeColor = getRouteColorBasedOnSettings(route.operator, route)
 
     return (
-      <div className="route-row" role="table">
+      <div className="route-row" role="group">
         {/* header row */}
-        <div className="header" role="row">
+        <div
+          className="header"
+          style={{
+            backgroundColor: routeColor,
+            color: getMostReadableTextColor(routeColor, route?.textColor)
+          }}
+        >
           {/* route name */}
           <div className="route-name">
-            <strong>
-              <RouteRenderer leg={generateFakeLegForRouteRenderer(route)} />
+            <strong
+              style={
+                routeName && routeName?.length >= 4 ? { fontSize: '3vb' } : {}
+              }
+            >
+              <div
+                style={{
+                  alignContent: 'center',
+                  display: 'flex',
+                  justifyContent: 'center',
+                  whiteSpace: 'nowrap'
+                }}
+              >
+                {showOperatorLogos && (
+                  <OperatorLogo operator={route?.operator} />
+                )}
+                <RouteRenderer
+                  // All GTFS bg colors look strange with the top border
+                  hideTopBorder={route?.operator?.colorMode?.includes('gtfs')}
+                  leg={generateFakeLegForRouteRenderer(route, true)}
+                />
```

Comment:
Sounds like you need another RouteRenderer component.

Reply:
I'm not sure I understand what you mean. This RouteRenderer will be whichever one is defined in the config